src/db.c

```diff
@@ -332,11 +381,38 @@ robj *dbRandomKey(redisDb *db) {
     }
 }
 
+/* Return random non-empty dictionary from this DB. */
+dict *getRandomDict(redisDb *db) {
+    if (db->dict_count == 1) return db->dict[0];
+    unsigned long long target = rand();
+    /* If number of keys is greater than max target value, then we should call target once more and get more entropy. */
+    if (db->key_count > RAND_MAX) {
+        target <<= 32;
+        target |= rand();
+    }
```

Comment:
Don't we have a 64bit random in mt199937-64? I believe that is as fast as rand.

Reply:
It would be nice, let me check that.

Reply:
changed to randomULong, I think that should be sufficient.